### PR TITLE
Change course language

### PR DESCRIPTION
**Pull request by Maijjay**
Screenshot from 2023-03-08 17-56-10
Screenshot from 2023-03-08 17-56-49

### CODE REVIEW

**Comment by george-misan** [comment]
Everything looks fine, But I think they should be a border around the flag icon (as depicted in the image below) because the Finland flag looks weird with no border. I will suggest using 2px of border width and color: baseTheme.colors.grey[200]
image

**Comment by Maijjay** [comment]
Here are the borders for the flags:
Screenshot from 2023-03-14 22-48-12
Screenshot from 2023-03-14 22-48-03

**Review by nygrenh — CHANGES_REQUESTED** [review]

File: services/headless-lms/server/src/controllers/course_material/courses.rs [hunk 1]
```diff
@@ -621,6 +621,42 @@ async fn get_public_top_level_pages(
...
+) -> ControllerResult<web::Json<Vec<Course>>> {
+    let mut conn = pool.acquire().await?;
+    let token = skip_authorize()?;
+    let course = models::courses::get_course(&mut conn, *course_id).await?;
+    let language_versions =
+        models::courses::get_all_language_versions_of_course(&mut conn, &course).await?;
+    token.authorized_ok(web::Json(language_versions))
+}
+
+#[generated_doc]
```
Comment (nygrenh) [inline services/headless-lms/server/src/controllers/course_material/courses.rs:641]: Add a documentation commment for endpoint path like in line 625

File: services/headless-lms/server/src/controllers/course_material/courses.rs [hunk 2]
```diff
@@ -666,5 +705,10 @@ pub fn _add_routes(cfg: &mut ServiceConfig) {
         .route(
             "/{course_id}/references",
             web::get().to(get_material_references_by_course_id),
-        );
+        )
+        .route(
+            "/{course_id}/pages/{page_language_group_id}",
```
Comment (nygrenh) [inline services/headless-lms/server/src/controllers/course_material/courses.rs:710]:
```suggestion
            "/{course_id}/pages/by-language-group-id/{page_language_group_id}",
```

File: services/headless-lms/models/src/pages.rs [hunk 3]
```diff
@@ -2840,6 +2908,40 @@ WHERE pages.id = $1
...
+    title,
+    deleted_at,
+    content,
+    order_number,
+    copied_from,
+    hidden,
+    page_language_group_id
+FROM pages p
+WHERE p.course_id = $1
+    AND p.page_language_group_id = $2
```
Comment (nygrenh) [inline services/headless-lms/models/src/pages.rs:2935]: AND deleted_at is NULL

**Review by nygrenh — CHANGES_REQUESTED** [review]

File: services/headless-lms/server/src/controllers/course_material/courses.rs (hunk 1 above)
Comment (nygrenh) [reply services/headless-lms/server/src/controllers/course_material/courses.rs:644]: Also one sentence here

**Review by nygrenh — CHANGES_REQUESTED** [review]

**Review by nygrenh — CHANGES_REQUESTED** [review]
Does this advertise a language if the language course is still a draft?

**Pull request merged**